python/ray/worker.py

```diff
@@ -13,6 +13,7 @@
 import os
```

Comment:
I'm not sure how easy this would be, but instead of modifying each API method within `worker.py`, what would be nice is to simply change `global_worker` to be equal to a `Client` object instead of a `Worker` object. What goes in/out of the `Worker` class is a bit blurry, so this could be difficult, but I think that would be the cleanest approach.

Reply:
Agree. I hadn't done this way since I'm not 100% familiar with all the edge cases ```Worker``` handles. We already skip a lot of the validation and error handling. Have there been more discussions about having a ```Worker``` base class that others derive from?

Reply:
We haven't had much discussion about it, but it makes sense to move in that direction. It's too much to do in this PR I think, so we can save that for after.

Reply:
Yeah, I agree that it shouldn't go in this PR. But it may be worth discussing more in some other thread since the move to support more languages may want to push this abstraction further down than Python.

I hesitate to set ```global_worker``` to ```Client``` outright given the complexity and fact we might sweep aside anyway. For example, if we want to minimize changes to the code in the public-facing API (e.g., ```wait```), we'd need ```Client``` to implement functions and logic that ```Worker``` has (and I don't have your accumulated bug experience!). It is unpleasantly gnarly to interleave ```Client``` code in ```Worker```, though, so let me know what you'd prefer to do.